Planck rev6 "TERMINAL_ENABLE = yes" breaks build

If you put `TERMINAL_ENABLE = yes` in rules.mk for planck rev6, build breaks with following error:

```
$ make planck/rev6:default
QMK Firmware 0.6.159
Making planck/rev6 with keymap default

arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 7-2018-q2-update) 7.3.1 20180622 (release) [ARM/embedded-7-branch revision 261907]
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Compiling: keyboards/planck/rev6/matrix.c                                                           [OK]
Compiling: quantum/audio/muse.c                                                                     [OK]
Compiling: keyboards/planck/planck.c                                                                [OK]
Compiling: keyboards/planck/rev6/rev6.c                                                             [OK]
Compiling: keyboards/planck/keymaps/default/keymap.c                                                [OK]
Compiling: quantum/quantum.c                                                                        [OK]
Compiling: quantum/keymap_common.c                                                                  [OK]
Compiling: quantum/keycode_config.c                                                                 [OK]
Compiling: quantum/process_keycode/process_audio.c                                                  [OK]
Compiling: quantum/process_keycode/process_clicky.c                                                 [OK]
Compiling: quantum/audio/audio_arm.c                                                                [OK]
Compiling: quantum/audio/voices.c                                                                   [OK]
Compiling: quantum/audio/luts.c                                                                     [OK]
Compiling: quantum/process_keycode/process_music.c                                                  [OK]
Compiling: quantum/process_keycode/process_terminal.c                                              In file included from tmk_core/common/print.h:76:0,
                 from tmk_core/common/debug.h:22,
                 from quantum/keymap.h:35,
                 from quantum/quantum.h:29,
                 from quantum/process_keycode/process_terminal.h:20,
                 from quantum/process_keycode/process_terminal.c:17:
tmk_core/common/chibios/printf.h:108:16: error: conflicting types for 'tfp_printf'
 #define printf tfp_printf
                ^
tmk_core/common/chibios/printf.h:103:6: note: previous declaration of 'tfp_printf' was here
 void tfp_printf(char *fmt, ...);
      ^~~~~~~~~~
tmk_core/common/chibios/printf.h:109:17: error: conflicting types for 'tfp_sprintf'
 #define sprintf tfp_sprintf
                 ^
tmk_core/common/chibios/printf.h:104:6: note: previous declaration of 'tfp_sprintf' was here
 void tfp_sprintf(char* s,char *fmt, ...);
      ^~~~~~~~~~~
 [ERRORS]
 | 
 | 
 | 
make[1]: *** [tmk_core/rules.mk:359: .build/obj_planck_rev6_default/quantum/process_keycode/process_terminal.o] Error 1
Make finished with errors
make: *** [Makefile:544: planck/rev6:default] Error 1

```

--- a/common_features.mk
+++ b/common_features.mk
@@ -213,6 +213,7 @@ endif
 ifeq ($(strip $(TERMINAL_ENABLE)), yes)
     SRC += $(QUANTUM_DIR)/process_keycode/process_terminal.c
     OPT_DEFS += -DTERMINAL_ENABLE
+    OPT_DEFS += -DUSER_PRINT
 endif
 
 ifeq ($(strip $(USB_HID_ENABLE)), yes)

--- a/tmk_core/common.mk
+++ b/tmk_core/common.mk
@@ -43,8 +43,10 @@ ifeq ($(PLATFORM),CHIBIOS)
     TMK_COMMON_DEFS += -DSTM32_EEPROM_ENABLE
   else
     TMK_COMMON_SRC += $(PLATFORM_COMMON_DIR)/eeprom_teensy.c
-endif
+  endif
   ifeq ($(strip $(AUTO_SHIFT_ENABLE)), yes)
+    TMK_COMMON_SRC += $(CHIBIOS)/os/various/syscalls.c
+  else ifeq($(strip $(TERMINAL_ENABLE)), yes)
     TMK_COMMON_SRC += $(CHIBIOS)/os/various/syscalls.c
   endif
 endif

--- a/tmk_core/common/print.h
+++ b/tmk_core/common/print.h
@@ -73,7 +73,9 @@ void print_set_sendchar(int8_t (*print_sendchar_func)(uint8_t));
 
 #elif defined(PROTOCOL_CHIBIOS) /* PROTOCOL_CHIBIOS */
 
+#ifndef TERMINAL_ENABLE
 #  include "chibios/printf.h"
+#endif
 
 #  ifdef USER_PRINT /* USER_PRINT */
 

